Hyperscan: use hs_valid_platform() to test host support

jviiret opened the pull request:
Versions of [Hyperscan ](https://01.org/hyperscan) >= 4.4.0 have the function `hs_valid_platform()`, which can be used to query whether the host has the architecture features required to use Hyperscan (e.g. `SSSE3`).

This patch set uses this function (on recent enough Hyperscan versions) to make a Snort binary built with Hyperscan support fail gracefully if the system does not have the required features.

snortadmin commented:
Thanks, we will run this through our integration process.  I'll update you.

snortadmin commented:
We reviewed this internally and would like to see a couple changes:

1.  Require hs >= 4.4.0 to avoid starting support for various versions at this point.
2.  Don't store hyperscan_available in SnortConfig.  Just compute as needed.

We are considering changing the plugin API (eg possibly changing the plugin init funcs to return bool) so this can be done once per plugin upon dynamic load to prevent further use.

The API change is on us.  If you want to address 1 and 2 above that would be great otherwise we can do that too.

Thanks!

jviiret commented:
No worries. I'll make those changes and update this PR!

jviiret commented:
I have updated and rebased these commits for changes 1 and 2 above here, it's a very small patch now.

One wrinkle: I have reworked autotools/cmake detection to require Hyperscan >= 4.4.0 when using `pkg-config`, as we include our version information in the `.pc` file and it's easy to do.

If the user points Snort manually at an installation of Hyperscan using `--with-hyperscan-libraries` etc, however, I don't have an easy way to do the appropriate checks. Hyperscan doesn't have a way to check for its version other than building and running a test program that parses the version string produced by the `hs_version()` API function, which is a lot of extra configuration tooling. 

(I'll have a talk to the team about adding some version #defines to the headers in future Hyperscan releases)

snortadmin commented:
This branch was run through our integration server internally, updated to include documentation, and merged there.  I'm going to close this.  Thanks!
